pkg/controller/migmigration/hooks.go

```diff
@@ -0,0 +1,212 @@
+package migmigration
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	migapi "github.com/konveyor/mig-controller/pkg/apis/migration/v1alpha1"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func (t *Task) runHooks() (int32, error) {
+	var hookRef *corev1.ObjectReference
+	var client k8sclient.Client
+	var err error
+
+	switch t.Phase {
+	case PreBackupHooks:
+		hookRef = t.PlanResources.MigPlan.Spec.Hooks.PreBackup
+	case PostBackupHooks:
+		hookRef = t.PlanResources.MigPlan.Spec.Hooks.PostBackup
+	case PreRestoreHooks:
+		hookRef = t.PlanResources.MigPlan.Spec.Hooks.PreRestore
+	case PostRestoreHooks:
+		hookRef = t.PlanResources.MigPlan.Spec.Hooks.PostRestore
+	default:
+		err := fmt.Errorf("Unknown Hook Phase %s", t.Phase)
+		log.Trace(err)
+		return -1, err
+	}
+
+	hook := migapi.MigHook{}
+	job := &batchv1.Job{}
+
+	if hookRef != nil {
+		err = t.Client.Get(
+			context.TODO(),
+			types.NamespacedName{
+				Name:      hookRef.Name,
+				Namespace: hookRef.Namespace,
+			},
+			&hook)
+		if err != nil {
+			log.Trace(err)
+			return -1, err
+		}
+
+		switch hook.Spec.TargetCluster {
+		case "destination":
+			client, err = t.getDestinationClient()
+			if err != nil {
+				log.Trace(err)
+				return -1, err
+			}
+		case "source":
+			client, err = t.getSourceClient()
+			if err != nil {
+				log.Trace(err)
+				return -1, err
+			}
+		default:
+			err := fmt.Errorf("targetCluster myst be 'source' or 'destination'. %s unknown", hook.Spec.TargetCluster)
+			log.Trace(err)
+			return -1, err
+		}
+
+		if hook.Spec.Playbook == "" {
+			job = t.baseJobTemplate(hook)
```

Comment:
Wouldn't it mean that the phase could be skipped at this point? Like
```golang
return 1, nil
```

Reply:
No. If the playbook is empty we run what we presume to be a custom container that may or may not contain ansible. Testing I have some containers that just run sleep, ls, oc get pods, etc. Could be bash, python, go whatever. We don't ask questions we just let the pod run with its configured entrypoint.